baseplate/lib/metrics.py

```diff
@@ -249,7 +259,7 @@ def counter(self, name: str) -> "Counter":
         counter_name = _metric_join(self.namespace, name.encode("ascii"))
         batch_counter = self.counters.get(counter_name)
```

Comment:
i think we need to figure out how to handle tags here. right now, the first metric of a given name will apply its tags to all subsequent ones. e.g. "foo#a=b" and "foo#x=y" will collide because we're only storing the BatchCounter objects based on metric name not tag

Reply:
flatten dictionary and use in key for self.counters